### PR TITLE
Added spinner to OntReception and minor refactor

### LINKED ISSUE #466: GPL-518 As a developer I would like to add feedback to the reception screen when plates are being imported to let users know that something is happening
**User story**
As a developer I would like to add feedback to the reception screen when plates are being imported to let users know that something is happening

**Who are the primary contacts for this story**
Steve / Harry

**Acceptance criteria**
To be considered successful the solution must allow:
- when plates are being loaded there should be a loading spinner
- disable the import button so users don't try and press it again

**Dependencies**
This story is blocked by the following dependencies:
- #<issue_no.>
- sanger/<repo>#<issue_no.>

**Additional context**

### PR DESCRIPTION
**Pull request by harrietc52**
Closes #466

### CODE REVIEW

File: src/views/ont/OntReception.vue [hunk 1]
```diff
@@ -27,50 +31,53 @@ import Alert from '@/components/Alert'
...
+    Spinner
   },
   methods: {
-    async getSequencescapePlates (barcodes) {
-      return await getPlates(this.$store.getters.api.sequencescape.plates, barcodes)
-    },
-    getBarcodes () {
-      return this.barcodes.split('\n').filter(Boolean).join(',')
+    async getSequencescapePlates() {
+      await getPlates(this.sequencescapeRequest, this.formattedBarcodes)
```
Comment (stevieing) [inline src/views/ont/OntReception.vue:52]: I think this needs a return

File: src/views/ont/OntReception.vue [hunk 2]
```diff
@@ -27,50 +31,53 @@ import Alert from '@/components/Alert'
...
     },
-    async handleSequencesapePlates () {
-      let jsonPlates = await this.getSequencescapePlates(this.getBarcodes())
-      if (jsonPlates !== undefined) {
-        this.plates = transformPlates(jsonPlates)
+    async handleCreateTractionPlates () {
+      this.busy = true
+      let ssPlatesJson = await this.getSequencescapePlates()
+
+      if (ssPlatesJson.length === 0) {
```
Comment (stevieing) [inline src/views/ont/OntReception.vue:58]: I think this should be 
`if (ssPlatesJson.length === undefined) {`

File: src/views/ont/OntReception.vue (hunk 1 above)
Comment (harrietc52) [reply src/views/ont/OntReception.vue:52]: Oh interesting thanks, I couldnt test with valid barcodes becuase the plate I was generating didnt have tagged samples, so might need you to test again after changes! But ill add that return :)

File: src/views/ont/OntReception.vue (hunk 2 above)
Comment (stevieing) [reply src/views/ont/OntReception.vue:58]: Is it possible to add a test for it? You could add a test to say that when ssPlatesJson is undefined it sets busy to false.

File: src/views/ont/OntReception.vue (hunk 1 above)
Comment (stevieing) [reply src/views/ont/OntReception.vue:52]: Again could we add a test for this?

Comment (harrietc52) [reply src/views/ont/OntReception.vue:52]: Oh Its a method, not a computed, so dont think it needs the return?

File: src/views/ont/OntReception.vue (hunk 2 above)
Comment (harrietc52) [reply src/views/ont/OntReception.vue:58]: Adding the return to the above returns an empty array, so will keep the condition the same

File: src/views/ont/OntReception.vue (hunk 1 above)
Comment (harrietc52) [reply src/views/ont/OntReception.vue:52]: Ah I remember trying, but struggled as getPlates is being imported in from the SequencescapePlates api, so couldnt mock it.

Comment (harrietc52) [reply src/views/ont/OntReception.vue:52]: Ah I remember now trying, but couldnt as getPlates is imported so struggled to mock it

**Pull request merged**